source/simulator/helper_functions.cc

```diff
@@ -1805,27 +1805,36 @@ namespace aspect
   double
   Simulator<dim>::compute_initial_newton_residual(LinearAlgebra::BlockVector &linearized_stokes_initial_guess)
   {
+    // Store the values of the current_linearization_point and linearized_stokes_initial_guess so we can reset them again.
     LinearAlgebra::BlockVector temp_linearization_point = current_linearization_point;
+    LinearAlgebra::BlockVector temp_linearized_stokes_initial_guess = linearized_stokes_initial_guess;
```

Comment:
I am confused about two things in this function:
- `linearized_stokes_initial_guess` after the function will be the same as before the function (you reset it at the end). Can it be `const` and you hand over `temp_linearized_stokes_initial_guess` to `denormalize_pressur` instead?
- You are only interested in the `system_rhs` at least that is the only result from the call to `assemble_stokes_system` that seems to be used. Why do we need to assemble the matrix for that?